lib/reversetunnel/localsite.go

```diff
@@ -90,10 +113,93 @@ func (s *localSite) GetLastConnected() time.Time {
 	return time.Now()
 }
 
-// Dial dials a given host in this site (cluster).
-func (s *localSite) Dial(from net.Addr, to net.Addr) (net.Conn, error) {
-	s.log.Debugf("local.Dial(from=%v, to=%v)", from, to)
-	return net.Dial(to.Network(), to.String())
+func (s *localSite) DialAuthServer() (net.Conn, error) {
+	var (
+		err  error
+		conn net.Conn
+	)
+
+	// get list of local auth servers
+	authServers, err := s.client.GetAuthServers()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// try and dial to one of them, as soon as we are successful, return the net.Conn
+	for _, authServer := range authServers {
+		conn, err = net.DialTimeout("tcp", authServer.GetAddr(), defaults.DefaultDialTimeout)
+		if err == nil {
+			return conn, nil
+		}
+	}
+
+	return nil, trace.ConnectionProblem(nil, "unable to connect to auth server")
```

Comment:
Information about the actual dial error is lost since you're ignoring `err`, is that on purpose?